httpmw/ip_cognito_auth.go

```diff
@@ -126,10 +126,9 @@ type CognitoClaims struct {
 // Note:
 // If the expiration duration is less than one, the items in the cache never expire (by default), and must be deleted manually.
 // If the cleanup interval is less than one, expired items are not deleted from the cache.
-func IpCognitoAuth(ipRange string, svc cognitoidentityprovideriface.CognitoIdentityProviderAPI, clientID string, expire time.Duration) gin.HandlerFunc {
+func IpCognitoAuth(ipRange string, svc cognitoidentityprovideriface.CognitoIdentityProviderAPI, clientID string, cache redis.Cmdable, expire time.Duration) gin.HandlerFunc {
```

Comment:
I would suggest re-arranging the arguments, dependencies first, and then arguments, something like:
```go
func IpCognitoAuth(svc cognitoidentityprovideriface.CognitoIdentityProviderAPI, cache redis.Cmdable, clientID string, ipRange string, expire time.Duration) gin.HandlerFunc {
  // awesome function goes here...
}
```